src/services/MakeCsv.php

```diff
@@ -31,12 +38,54 @@ public function getFileName(): string
         return date('Y-m-d') . '-export.elabftw.csv';
     }
 
+ 
+    /**
+     * Populate the entitiesData and metadataColumns arrays
+     */
+    protected function populateEntitiesAndMetadata()
+    {
+        $mdPositions = array();
+        foreach ($this->idArr as $id) {
+           $this->Entity->setId((int) $id);
+           try {
+               $permissions = $this->Entity->getPermissions();
+           } catch (IllegalActionException $e) {
+               continue;
+           }
+           if ($permissions['read']) {
+               $entityData = $this->Entity->entityData;
+               $entityData['url'] = $this->getUrl();
+               $metadataValues = array();
+               if ($entityData['metadata']) {
+                   $decoded = json_decode($entityData['metadata'], true, 512, JSON_THROW_ON_ERROR);
+                   if ($decoded && isset($decoded['extra_fields']) && is_array($decoded['extra_fields'])) {
+                       foreach ($decoded['extra_fields'] as $field => $details) {
+                           $mdPositions[$field] = intval($details['position']);
+                           $metadataValues[$field] = $details['value'] ?? '';
+                       }
+                       
+                   }
+               } 
+               $entityData['metadata_values'] = $metadataValues;
+               $this->entitiesData[] = $entityData;
+           }
+        }
+        asort($mdPositions);
+        $this->metadataColumns = array_keys($mdPositions);
+    }
+
+    
     /**
      * Here we populate the first row: it will be the column names
      */
     protected function getHeader(): array
     {
-        return  array('id', 'date', 'title', 'content', 'category', 'elabid', 'rating', 'url', 'metadata');
+       $this->populateEntitiesAndMetadata();
+       $standardColumns = array('id', 'date', 'title', 'content', 'category', 'elabid', 'rating', 'url', 'metadata');
+       // in case of conflicting column names, add "metadata_" prefix to metadata columns
```

Comment:
I wonder if all the columns coming from metadata should not be prefixed. This way during the import we can add a rule that all metadata_ prefixed columns are imported as metadata. This would also allow users to choose which columns they want to see populated as metadata!